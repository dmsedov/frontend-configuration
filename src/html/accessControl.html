<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  </head>
  <body>
    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#exampleModalCenter">
      Запустить модальное окно
    </button><br><br>
    <div class="modal fade" id="exampleModalCenter" tabindex="-1" role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
      <div class="modal-dialog modal-dialog-centered modal-lg" role="document">
        <div class="modal-content">
          <div class="modal-header">
            <h5 class="modal-title" id="exampleModalLongTitle">Modal title</h5>
            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div class="modal-body d-flex justify-content-center">
            <form class="access-form d-flex flex-column" id="access-user-form">
              <div class="user-rights-fields d-flex justify-content-around">
                <div class="form-check-inline user-name"><span>Пользователь</span></div>
                <div class="user-rights">
                    <span>Чтение</span>
                </div>
                <div class="user-rights">
                    <span>Запись и чтение</span>
                </div>
                <div class="delete-user">Удалить пользователя</div>
              </div>
              <div class="form-check form-check-inline user-rights-fields d-flex justify-content-around">
                <label class="form-check-label user-name">Петя</label>
                <input class="form-check-input user-rights" type="radio" name="Петя" value="r" checked>
                <input class="form-check-input user-rights" type="radio" name="Петя" value="rw">
                <button type="button del-user" class="delete-user">Удалить</button>
              </div>
              <div class="form-check form-check-inline user-rights-fields d-flex">
                <label class="form-check-label">Дима</label>
                <input class="form-check-input" type="radio" name="Дима" value="r" checked>
                <input class="form-check-input" type="radio" name="Дима" value="rw">
                <button type="button delete-user" class="delete-user">Удалить</button>
              </div>
              <div class="form-check form-check-inline user-rights-fields d-flex ">
                <label class="form-check-label">Женя</label>
                <input class="form-check-input" type="radio" name="Женя" value="r" checked>
                <input class="form-check-input" type="radio" name="Женя" value="rw">
                <button type="button del-user" class="delete-user">Удалить</button>
              </div>
              <div class="form-controls">
                <button type="submit" class="btn btn-primary">Сохранить</button>
                <button type="button" class="btn btn-secondary" data-dismiss="modal">Закрыть</button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>
  </html>
