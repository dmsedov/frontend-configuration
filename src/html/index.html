<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  </head>
  <body>
    <div id="container">
      <header>
        <div class="menu-position">
          <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
            <a class="navbar-brand" href="#">Calendula</a>
            <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
              <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="navbarNav">
              <ul class="navbar-nav">
                <li class="nav-item active">
                  <a class="nav-link" href="#">Сгенерировать ссылку <span class="sr-only">(current)</span></a>
                </li>
                <li class="nav-item">
                  <a class="nav-link" href="#">Настройки доступа</a>
                </li>
                <li class="nav-item">
                  <a class="nav-link" href="#">Выйти</a>
                </li>
              </ul>
            </div>
          </nav>
        </div>
      </header>
      <main>
        <div class="calendar" style="height: 90vh">

        </div>
      </main>
      <div></div>
    </div>
  </body>
</html>
