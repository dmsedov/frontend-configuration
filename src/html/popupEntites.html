<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  </head>
  <body class="modal-open">
    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#access-control">
      Запустить модальное окно Access contol form
    </button><br><br>
    <div class="modal fade" id="access-control" tabindex="-1" role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
      <div class="modal-dialog modal-dialog-centered modal-lg" role="document">
        <div class="modal-content">
          <div class="modal-header">
            <h5 class="modal-title" id="users-control">Управление правами пользователей</h5>
            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div class="modal-body d-flex justify-content-center">
            <form class="access-form d-flex flex-column" id="access-user-form">
              <div class="user-rights-fields d-flex justify-content-around align-items-center">
                <div class="form-check-inline user-name"><span>Nickname</span></div>
                <div class="user-rights">Чтение</div>
                <div class="user-rights">Запись и чтение</div>
                <div class="act-delete-user">Удалить</div>
              </div>
              <div class="form-check form-check-inline user-rights-fields d-flex">
                <div class="form-check-inline user-name">
                  <label>Петя</label>
                </div>
                <div class="user-rights">
                  <input type="radio" name="Петя" value="r" checked>
                </div>
                <div class="user-rights">
                  <input type="radio" name="Петя" value="rw">
                </div>
                <div class="act-delete-user">
                  <button type="button" class="btn btn-primary btn-sm">Удалить</button>
                </div>
              </div>
              <div class="form-check form-check-inline user-rights-fields d-flex">
                <div class="form-check-inline user-name">
                  <label>Женя</label>
                </div>
                <div class="user-rights">
                  <input type="radio" name="Женя" value="r" checked>
                </div>
                <div class="user-rights">
                  <input type="radio" name="Женя" value="rw">
                </div>
                <div class="act-delete-user">
                  <button type="button" class="btn btn-primary btn-sm">Удалить</button>
                </div>
              </div>
              <div class="form-check form-check-inline user-rights-fields d-flex">
                <div class="form-check-inline user-name">
                  <label>Дима</label>
                </div>
                <div class="user-rights">
                  <input type="radio" name="Дима" value="r" checked>
                </div>
                <div class="user-rights">
                  <input type="radio" name="Дима" value="rw">
                </div>
                <div class="act-delete-user">
                  <button type="button" class="btn btn-primary btn-sm">Удалить</button>
                </div>
              </div>
              <div class="form-controls">
                <button type="submit" class="btn btn-primary btn-sm">Сохранить</button>
                <button type="button" class="btn btn-secondary btn-sm" data-dismiss="modal">Закрыть</button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <br>
    <br>
    <br>
    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#entry-link">
      Запустить модальное окно Copy link to buffer
    </button>
    <br>
    <br>
    <div id="entry-link" class="modal" tabindex="-1" role="dialog">
      <div class="modal-dialog modal-dialog-centered" role="document">
        <div class="modal-content">
          <div class="modal-header">
            <h5 class="modal-title">Данные для доступа</h5>
            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div class="modal-body">
            <p class="description">Нажмите "Скопировать" для копирования ссылки в буфер</p>
            <input type="text" name="entryLink" readonly>
            <button type="button" class="btn btn-primary btn-sm">Скопировать</button>
          </div>
        </div>
      </div>
    </div>
    <br>
    <br>
    <div style="display:flex;justify-content: center;">
      <div class="auth-form">
        <h2>Авторизация</h2>
        <p class="auth-tip">Авторизуйтесь через представленные ниже сервисы:</p>
        <ul class="list-sm">
          <li class="auth-service">
            <a class="btn vk-sm"href="#">Вконтакте</a>
          </li>
          <li class="auth-service">
            <a class="btn google-sm" href="#">Google</a>
          </li>
        </ul>
      </div>
    </div>
    <br><br><br>
    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#editor">
      Запустить модальное окно Event editor form
    </button><br><br>
    <div class="modal show" id="editor" tabindex="-1" role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true" style="display:block;"> <!--class="show" style="display:block" для отображения модального окна; в тег body необходим класс "modal-open" для прокрутки содержимого окна-->
      <div class="modal-dialog" role="document">
        <div class="modal-content">
          <div class="modal-header">
            <h5 class="modal-title" id="event">Событие</h5>
            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div class="modal-body">
            <form class="event-form">
              <div class="controls">
                <button type="submit" class="btn btn-primary btn-sm">Сохранить и закрыть</button>
                <button type="button" class="btn btn-primary btn-sm">Удалить</button>
              </div>
              <div class="event-content">
                <input class="name-field" type="text" name="name" placeholder="Событие">
                <input class="place-field" type="text" name="place" placeholder="Место">
                <textarea class="description-field" name="text" rows="20" placeholder="Описание"></textarea>
              </div>
            </form>
            <div class="authors">
              <h5>Авторы</h5>
              <ul class="authors-list">
                <li>Дмитрий Седов</li>
                <li>Василий Пупкин</li>
                <li>Яков Останин</li>
                <li>Артемий Хованский</li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>
</html>
