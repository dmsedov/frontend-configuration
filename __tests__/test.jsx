import 'js-polyfills/html';

import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MyForm from '../src/js/components/Form';

Enzyme.configure({ adapter: new Adapter() });

test('MyForm', () => {
  const wrapper = mount(<MyForm />);
  expect(wrapper.render()).toMatchSnapshot();

  const input = wrapper.find('form');
  input.simulate('submit');
  expect(wrapper.render()).toMatchSnapshot();
});
