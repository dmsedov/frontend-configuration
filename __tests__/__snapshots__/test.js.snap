// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyForm 1`] = `
<form>
  <div
    class="form-row"
  >
    <div
      class="form-group col-md-6"
    >
      <label
        class="col-form-label"
        for="inputEmail4"
      >
        Email
      </label>
      <input
        class="form-control"
        id="inputEmail4"
        placeholder="Email"
        type="email"
        value=""
      />
    </div>
    <div
      class="form-group col-md-6"
    >
      <label
        class="col-form-label"
        for="inputPassword4"
      >
        Password
      </label>
      <input
        class="form-control"
        id="inputPassword4"
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    class="form-group"
  >
    <label
      class="col-form-label"
      for="inputAddress"
    >
      Address
    </label>
    <textarea
      class="form-control"
      id="inputAddress"
      placeholder="1234 Main St"
      type="text"
    />
  </div>
  <div
    class="form-row"
  >
    <div
      class="form-group col-md-6"
    >
      <label
        class="col-form-label"
        for="inputCity"
      >
        City
      </label>
      <input
        class="form-control"
        id="inputCity"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group col-md-6"
    >
      <label
        class="col-form-label"
        for="inputCountry"
      >
        Country
      </label>
      <select
        class="form-control"
        id="inputCountry"
      >
        <option>
          Choose
        </option>
        <option
          value="argentina"
        >
          Argentina
        </option>
        <option
          value="russia"
        >
          Russia
        </option>
        <option
          value="china"
        >
          China
        </option>
      </select>
    </div>
  </div>
  <div
    class="form-group"
  >
    <div
      class="form-check"
    >
      <label
        class="form-check-label"
        for="rules"
      >
        <input
          class="form-check-input"
          id="rules"
          type="checkbox"
        />
        Accept Rules
      </label>
    </div>
  </div>
  <button
    class="btn btn-primary"
    type="submit"
  >
    Sign in
  </button>
</form>
`;

exports[`MyForm 2`] = `
<a
  href="#"
>
  Back
</a>
`;
